core/iwasm/libraries/lib-wasi-threads/test/stress_test_threads_creation.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2023 Amazon.com Inc. or its affiliates. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include <assert.h>
+#include <errno.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <stdbool.h>
+
+enum CONSTANTS {
+    NUM_ITER = 90000,
+    NUM_RETRY = 5,
+    MAX_NUM_THREADS = 8,
+    RETRY_SLEEP_TIME_US = 4000,
+};
+
+unsigned int threads_executed = 0;
+unsigned int threads_creation_tried = 0;
+
+pthread_mutex_t mutex;
+
+bool is_finished[MAX_NUM_THREADS];
+
+void *
+thread_func(void *arg)
+{
+    __atomic_fetch_add(&threads_executed, 1, __ATOMIC_RELAXED);
+
+    pthread_mutex_lock(&mutex);
+    int num = *(int *)(arg);
+    is_finished[num] = true;
+    pthread_mutex_unlock(&mutex);
+
+    return NULL;
+}
+
+void
+spawn_thread(pthread_t *thread, void *arg)
+{
+    int status_code = -1;
+    unsigned timeout = RETRY_SLEEP_TIME_US;
+    for (int tries = 0; status_code != 0 && tries < NUM_RETRY; ++tries) {
+        status_code = pthread_create(thread, NULL, &thread_func, arg);
+        ++threads_creation_tried;
+        assert(status_code == 0 || status_code == EAGAIN);
+        if (status_code == EAGAIN) {
+            usleep(timeout);
+            timeout *= 2;
+        }
+    }
+
+    if (status_code != 0) {
+        fprintf(stderr,
+                "Threads executed %d. Aborting because of failed retries.",
+                threads_executed);
+    }
+    assert(status_code == 0 && "Thread creation should succeed");
+}
+
+int
+main(int argc, char **argv)
+{
+    pthread_mutex_init(&mutex, NULL);
+    pthread_t threads[MAX_NUM_THREADS];
```

Comment:
Since you detach the thread, you probably don't need this array any more?

Reply:
hm, 

I had a concern that `pthread_t` var assigned to thread should be alive until thread finishes. 

I'm not a posix expert so I'm not 100% so tell me if I'm wrong

Reply:
hm,

I had a concern that pthread_t var assigned to thread should be alive until thread finishes.

I'm not a posix expert so I'm not 100% sure so tell me if I'm wrong

Reply:
You don't need to keep track on the thread I'd if you don't use it. `pthread_t` in wasi libc (and many other implementations) is simply an integer. Also, C doesn't have RAII by default soeven if it was a complex data structure allowing thread management, worse case you'd have a memory leak, but thread wouldn't be affected unless some function was explicitly called

Reply:
thanks. Got it